sdk/domain_authentication.go

```diff
@@ -153,17 +154,42 @@ func (c *Client) ValidateDomainAuthentication(ctx context.Context, id string) Re
 		}
 	}
 
-	_, statusCode, err := c.Post(ctx, "POST", "/whitelabel/domains/"+id+"/validate", nil)
+	respBody, statusCode, err := c.Post(ctx, "POST", "/whitelabel/domains/"+id+"/validate", nil)
 	if err != nil || statusCode != 200 {
 		return RequestError{
 			StatusCode: statusCode,
 			Err:        err,
 		}
 	}
 
+	res, err := unmarshall(respBody)
+	if err != nil {
+		return RequestError{
+			StatusCode: http.StatusInternalServerError,
+			Err:        err,
+		}
+	}
+
+	if valid, ok := res["valid"].(bool); ok && !valid {
+		return RequestError{
+			StatusCode: http.StatusInternalServerError,
+			Err:        ErrDomainAuthenticationValidationFailed,
+		}
+	}
+
 	return RequestError{StatusCode: http.StatusOK, Err: nil}
 }
 
+func unmarshall(respBody string) (map[string]interface{}, error) {
+	var j map[string]interface{}
+	if err := json.Unmarshal([]byte(respBody), &j); err != nil {
+		log.Printf("JSON Unmarshalling of the response body: %s, failed with: %s,",
```

Comment:
Not a thing we need to care about or worth spending time on, but does the `err` type contain the response body?

Reply:
The `err` just has the unmarshalling error.
Example: I tried a unit test with the response body of invalid json `alkdjfalkdjsf` the `err= invalid character 'a' looking for beginning of value`
```
terraform-provider-sendgrid/sdk git:err-if-response-is-not-valid
❯ go test
2023/07/14 14:32:38 JSON Unmarshalling of the response body: alkdjfalkdjsf, failed with: invalid character 'a' looking for beginning of value,
--- FAIL: TestClient_ValidateDomainAuthentication_ShouldErrorForResponseBodySayingValidFalse (0.00s)
    domain_authentication_test.go:180: ValidateDomainAuthentication() got = invalid character 'a' looking for beginning of value, want domain authentication validation failed
FAIL
```